org/mozilla/jss/ssl/javax/JSSParameters.java

```diff
@@ -190,4 +198,27 @@ public String getHostname() {
     public void setHostname(String server_hostname) {
         hostname = server_hostname;
     }
+
+    public void setApplicationProtocols(String[] protocols) throws IllegalArgumentException {
+        if (protocols == null) {
+            appProtocols = null;
+            return;
+        }
+
+        int index = 0;
+        for (String protocol : protocols) {
+            if (protocol != "" && protocol.getBytes(StandardCharsets.UTF_8).length > 255) {
```

Comment:
should ensure non-empty and length < 255, e.g.
```java
for (String protocol : protocols) {
    if (null == protocol)
        throw new IllegalArgumentException("null protocol");
    byte[] bs = protocol.getBytes(StandardCharsets.UTF_8);
    if (bs.length < 1 || bs.length > 255)
        throw new IllegalArgumentException("protocol length must be 1..255");
}
```

This is actually stricter than what's in the openjdk implementation (it has null/empty check but not < 255 check).
But that is important to check.